src/avram/callbacks.cr

```diff
@@ -80,6 +120,65 @@ module Avram::Callbacks
     end
   end
 
+  # Run the given method after `run` is called on an `Operation`.
+  # The return value of the `run` method is passed to `method_name`.
+  #
+  # ```
+  # after_run :log_entry
+  #
+  # private def log_entry(value)
+  #   log_stuff(value)
+  # end
+  # ```
+  macro after_run(method_name)
+    def after_run(object)
+      {% if @type.methods.map(&.name).includes?(:after_run.id) %}
+        previous_def
+      {% else %}
+        super
+      {% end %}
+
+      {{ method_name.id }}(object)
+    end
+  end
+
+  # Run the given block after the operation runs
+  #
+  # The return value from `run` will be passed to this block.
+  #
+  # ```
+  # class GenerateReport < Avram::Operation
+  #   after_run do |value|
+  #     value == "some report"
+  #   end
+  #
+  #   def run
+  #     "some report"
+  #   end
+  # end
+  # ```
+  macro after_run(&block)
+    {%
+      if block.args.size != 1
+        raise <<-ERR
+        The 'after_run' callback requires exactly 1 block arg to be passed.
+        Example:
+          after_run { |value| some_method(value) }
+        ERR
+      end
+    %}
+    def after_run(object)
```

Comment:
```suggestion
    def after_run(%object)
```

If users pass a block like`{ |object| do_something(object) }` it fails to compile (the variable usage will have to be updated as well)

Reply:
I'm not sure that I follow here. Also, doesn't `%object` just turn it in to a fresh var so you wouldn't have a `object` local variable available to you?

Reply:
Ohh.. ok.. I'm waking up 😅  I see what you mean now.